### PR TITLE
viceroy: init at 0.3.5

### PR DESCRIPTION
**Pull request by ereslibre**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

/cc @shyim @nomeata

### CODE REVIEW

File: pkgs/development/tools/viceroy/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  cargoTestFlags = [
+    "--package viceroy-lib"
+  ];
+
+  meta = with lib; {
+    description = "Viceroy provides local testing for developers working with Compute@Edge";
+    homepage = "https://github.com/fastly/Viceroy";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ereslibre ];
```
Comment (ereslibre) [inline pkgs/development/tools/viceroy/default.nix:26]: @shyim if you want to be a maintainer let me know

**Comment by GabrielDougherty** [comment]
Result of `nixpkgs-review pr 220051` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>viceroy</li>
  </ul>
</details>

[nix-shell:~/.

**Review by GabrielDougherty — APPROVED** [review]

**Review by shyim — APPROVED** [review]

File: pkgs/development/tools/viceroy/default.nix (hunk 1 above)
Comment (shyim) [reply pkgs/development/tools/viceroy/default.nix:26]: feel free to add me :)

Comment (ereslibre) [reply pkgs/development/tools/viceroy/default.nix]: Done :)

**Review by shyim — APPROVED** [review]

**Review by shyim — APPROVED** [review]

File: pkgs/development/tools/viceroy/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ rustPlatform, fetchFromGitHub, lib, stdenv, Security }:
```
Comment (AndersonTorres) [inline pkgs/development/tools/viceroy/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub 
, rustPlatform
, Security 
}:
```

File: pkgs/development/tools/viceroy/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "viceroy";
+  version = "0.3.5";
+
+  src = fetchFromGitHub {
+    owner = "fastly";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/development/tools/viceroy/default.nix:9]:
```suggestion
    repo = "viceroy";
```

File: pkgs/development/tools/viceroy/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  buildInputs = lib.optional stdenv.isDarwin Security;
+
+  cargoHash = "sha256-vbhBlfHrFcjtaUJHYvB106ElYP0NquOo+rgIx9cWenY=";
+
+  cargoTestFlags = [
+    "--package viceroy-lib"
+  ];
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/development/tools/viceroy/default.nix:22]:
```suggestion
  meta = {
```

File: pkgs/development/tools/viceroy/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    description = "Viceroy provides local testing for developers working with Compute@Edge";
+    homepage = "https://github.com/fastly/Viceroy";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ereslibre shyim ];
+    platforms = platforms.unix;
```
Comment (AndersonTorres) [inline pkgs/development/tools/viceroy/default.nix:27]:
```suggestion
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ ereslibre shyim ];
    platforms = lib.platforms.unix;
```

File: pkgs/development/tools/viceroy/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  buildInputs = lib.optional stdenv.isDarwin Security;
+
+  cargoHash = "sha256-vbhBlfHrFcjtaUJHYvB106ElYP0NquOo+rgIx9cWenY=";
+
+  cargoTestFlags = [
+    "--package viceroy-lib"
+  ];
+
+  meta = with lib; {
+    description = "Viceroy provides local testing for developers working with Compute@Edge";
```
Comment (AndersonTorres) [inline pkgs/development/tools/viceroy/default.nix:23]:
```suggestion
    description = "A local testing for developers working with Compute@Edge";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Pull request merged**